shared/js/content-scripts/block-cookie.js

```diff
@@ -0,0 +1,80 @@
+// Set up 3rd party cookie blocker
+(function cookieBlocker () {
+    // don't inject into non-HTML documents (such as XML documents)
+    // but do inject into XHTML documents
+    if (document instanceof HTMLDocument === false && (
+        document instanceof XMLDocument === false ||
+        document.createElement('div') instanceof HTMLDivElement === false
+    )) {
+        return
+    }
+
+    function clearInjectedCookiesAndBlock () {
+        // Clear previously set cookies
+        var cookies = document.cookie.split('; ')
+        for (var c = 0; c < cookies.length; c++) {
+            var d = window.location.hostname
+            var cookieBase = encodeURIComponent(cookies[c].split(';')[0].split('=')[0]) + '=; expires=Thu, 01-Jan-1970 00:00:01 GMT; domain=' + d + ' ;path='
+            var p = location.pathname.split('/')
+            document.cookie = cookieBase + '/'
+            while (p.length > 0) {
+                document.cookie = cookieBase + p.join('/')
+                p.pop()
+            };
+            d.shift()
+        }
+
+        // disable setting cookies
+        document.__defineSetter__('cookie', function (value) { })
```

Comment:
```suggestion
        document.__defineSetter__('cookie', function (value) { })
        document.__defineGetter__('cookie', () => '')
```

When testing this code I saw that `document.cookie` somehow became `undefined`. This broke scripts in the iframe because, according to the spec `document.cookie` always returns a string. Adding a stubbed getter seems to fix this.